src/internal/task/etcd_service.go

```diff
@@ -118,17 +118,17 @@ func newEtcdDoer(namespaceEtcd *namespaceEtcd, group string, cache Cache) Doer {
 
 func (ed *etcdDoer) Do(ctx context.Context, inputChan chan *types.Any, cb CollectFunc) error {
 	return ed.withGroup(ctx, func(ctx context.Context, renewer *col.Renewer) error {
-		var eg errgroup.Group
 		prefix := path.Join(ed.group, uuid.NewWithoutDashes())
 		done := make(chan struct{})
 		var count int64
 		ctx, cancel := context.WithCancel(ctx)
+		eg, egCtx := errgroup.WithContext(ctx)
```

Comment:
I think we should just overwrite `ctx`. It is fine to skip the deletion when an error occurs because it will get cleaned up by the lease expiration. Also, for a bit of clarity, this isn't addressing a correctness issue, just ensuring fail-fast behavior when the watch independently errors?